metecho/api/jobs.py

```diff
@@ -1156,3 +1157,48 @@ def user_reassign(scratch_org, *, new_user, originating_user_id):
 
 
 user_reassign_job = job(user_reassign)
+
+
+def refresh_datasets(task: Task, user: User):
+    """
+    Refresh the dataset definition cache on `task` by parsing definitions from the Task branch
+    """
+    try:
+        task.refresh_from_db()
+        with local_github_checkout(user, task.get_repo_id(), task.branch_name):
```

Comment:
When we fetch datasets we do it from the HEAD of `task.branch_name` to get the latest definitions. This means sometimes the dataset definitions could be ahead of the schema on the Dev Org if the org is not running the HEAD commit. Should we somehow handle that?

Reply:
🤔

Reply:
At this point I can't wrap my head around all of the scenarios to know what's best. For now let's:

1. Warn the user if their extract declaration seems to refer to objects or fields that do not exist.
2. Ignore those elements if they choose to continue regardless

Reply:
At the suggestion of @davidmreed, we are now:

1. Warning the user if their org is out of date (not based on the latest commit)
2. Warning the user if there are fields/objects in the selected existing dataset that do not exist in the Org
3. Removing those outdated fields on save